extension/app/background.ts

```diff
@@ -1,3 +1,114 @@
+import { randomString } from "./src/lib/utils";
+import {
+  AUTH0_AUDIENCE,
+  AUTH0_CLIENT_DOMAIN,
+  AUTH0_CLIENT_ID,
+  AuthBackroundMessage,
+  Auth0AuthorizeResponse,
+  AuthBackgroundResponse,
+} from "./src/lib/auth";
+
+/**
+ * Listener to open/close the side panel when the user clicks on the extension icon.
+ */
 chrome.runtime.onInstalled.addListener(() => {
   void chrome.sidePanel.setPanelBehavior({ openPanelOnActionClick: true });
 });
+
+/**
+ * Listener for messages sent from the react app to the background script.
+ * For now we use messages to authenticate the user.
+ */
+chrome.runtime.onMessage.addListener(
+  (
+    message: AuthBackroundMessage,
+    sender,
+    sendResponse: (
+      response: Auth0AuthorizeResponse | AuthBackgroundResponse
+    ) => void
+  ) => {
+    switch (message.type) {
+      case "AUTHENTICATE":
+        authenticate(sendResponse);
```

Comment:
Shouldn't we await here?

Reply:
FWIW the linter config on `front` flags this, you might want to have it in this project as well.

Reply:
Nope we don't need to await. I added a void!

@tdraier added a linter yesterday, I'll add rules from front. Will do in a separate PR though 👍🏻